### PR TITLE
Create PledgedMerc Vault Token

### PR DESCRIPTION
**Pull request by 0xHannibal**
This needs some more testing, but wanted to get your feedback on how it
is looking so far. Big things to check out:

* Adapted solmate's 4626 to be OZ friendly
* Clone a PledgedMerc vault when creating a new gauge
* Restrict the gauge's pledge and depledge to the vault token

### CODE REVIEW

File: contracts/src/Gauge.sol [hunk 1]
```diff
@@ -91,6 +101,11 @@ contract Gauge is IERC721, ERC721Enumerable {
         totalWeight += mintPrice;
         mintPrice = mintPrice * 2;
 
+        string memory idStr = Strings.toString(id);
+        PledgedMerc pMerc = PledgedMerc(Clones.clone(address(defaultPledgedMerc)));
+        pMerc.initialize(this, id, string(abi.encodePacked("Pledged Merc Gauge ", idStr)), string(abi.encodePacked("pMERC-", idStr)));
```
Comment (tmpltn) [inline contracts/src/Gauge.sol]: `string.concat` is a nice new way to do this as of 0.8.12 or so.

File: contracts/src/Gauge.sol [hunk 2]
```diff
@@ -91,6 +101,11 @@ contract Gauge is IERC721, ERC721Enumerable {
...
+        pMercForGauges[id] = pMerc;
```
Comment (tmpltn) [inline contracts/src/Gauge.sol:107]: could also put this in the gauges struct

File: contracts/src/PledgedMerc.sol [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+     * @dev Returns the amount of Merc in this vault
+     */
+    function totalAssets() public override view returns (uint256) {
+        return pledgedMerc;
+    }
+
+    function beforeWithdraw(uint256 assets, uint256 shares, address owner) internal override {
+        shares;
+        pledgedMerc -= assets;
+        gauge.depledge(gaugeId, assets, owner);
```
Comment (tmpltn) [inline contracts/src/PledgedMerc.sol:60]: I guess we'll also need to have the Gauge contract check that the `gaugeId` can only be depledged by its pMERC contract.

File: contracts/src/PledgedMerc.sol [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    function afterDeposit(uint256 assets, uint256 shares, address receiver) internal override {
+        shares;
+        pledgedMerc += assets;
+        gauge.pledge(gaugeId, assets, receiver);
```
Comment (tmpltn) [inline contracts/src/PledgedMerc.sol:66]: same comment as above. Gauge will need to ensure only the vault can pledge.

**Review by tmpltn — COMMENTED** [review]
looks like the right direction for sure.

**Comment by tmpltn** [comment]
lgtm. feel free to merge when ready

File: contracts/src/Gauge.sol (hunk 1 above)
Comment (0xHannibal) [reply contracts/src/Gauge.sol]: TIL

File: contracts/src/PledgedMerc.sol (hunk 3 above)
Comment (0xHannibal) [reply contracts/src/PledgedMerc.sol:60]: Yeah, there is a `require` in the gauge contract

**Pull request merged**